Library/Homebrew/utils/analytics.rb

```diff
@@ -16,21 +16,26 @@ def report(type, metadata = {})
         args = %W[
           --max-time 3
           --user-agent #{HOMEBREW_USER_AGENT_CURL}
-          -d v=1
-          -d tid=#{ENV["HOMEBREW_ANALYTICS_ID"]}
-          -d cid=#{ENV["HOMEBREW_ANALYTICS_USER_UUID"]}
-          -d aip=1
-          -d an=#{HOMEBREW_PRODUCT}
-          -d av=#{HOMEBREW_VERSION}
-          -d t=#{type}
+          --data v=1
+          --data aip=1
+          --data-urlencode t=#{type}
+          --data-urlencode tid=#{ENV["HOMEBREW_ANALYTICS_ID"]}
+          --data-urlencode cid=#{ENV["HOMEBREW_ANALYTICS_USER_UUID"]}
+          --data-urlencode an=#{HOMEBREW_PRODUCT}
+          --data-urlencode av=#{HOMEBREW_VERSION}
         ]
-        metadata.each { |k, v| args << "-d" << "#{k}=#{v}" if k && v }
+        metadata.each do |key, value|
+          next unless key
+          next unless value
+          args << "--data-urlencode" << "#{key}=#{value}"
+        end
 
         # Send analytics. Don't send or store any personally identifiable information.
         # https://github.com/Homebrew/brew/blob/master/share/doc/homebrew/Analytics.md
         # https://developers.google.com/analytics/devguides/collection/protocol/v1/devguide
         # https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters
         if ENV["HOMEBREW_ANALYTICS_DEBUG"]
+          puts "#{ENV["HOMEBREW_CURL"]} #{args.join(" ")}"
```

Comment:
You will need to stick the URL (`https://www.google-analytics.com/debug/collect`) between the `curl` binary and the arguments if you want this output to match what is executed (and to match the code you added in the Bash version of this code).

Otherwise :+1: on adding this.